src/server.js

```diff
@@ -169,13 +169,15 @@ class Server extends EventEmitter {
                 handlerOpts = result;
                 handlerOpts.srcHead = head;
 
+                const data = { request, source: socket, head, handlerOpts, server: this };
```

Comment:
Would it make sense to make a class out of this instead of a plain object? It seems quite complex and we will be making an interface anyway. Just sayin'. 🤷

Reply:
Nope. Why a class?

Reply:
To encapsulate some logic which seems to be used in a lot of places and give it a name. Readability and documentation purposes. It's not important though. Just something that occured to me when I saw a plain object stuffed with 5 properties.

Reply:
Hmm... The documentation purposes is very convincing 🤔 I don't think it's worth it for 5 properties though. TypeScript interface should be sufficient to be honest.